liquigraph-core/src/test/java/org/liquigraph/core/io/ChangelogGraphWriterTestSuite.java

```diff
@@ -267,19 +275,26 @@ private static void assertThatChangesetIsStored(ResultSet resultSet) throws SQLE
         assertThat((Collection<String>) resultSet.getObject("queries"))
             .containsExactly("CREATE (n: SomeNode {text:'yeah'})");
 
-        Node changeset = (Node) resultSet.getObject("changeset");
-        assertThat(changeset.getProperty("id")).isEqualTo("identifier");
-        assertThat(changeset.getProperty("author")).isEqualTo("fbiville");
-        assertThat(changeset.getProperty("checksum")).isEqualTo(checksum(singletonList("CREATE (n: SomeNode {text:'yeah'})")));
+        Object changeset = resultSet.getObject("changeset");
+        assertThat(property(changeset, "id")).isEqualTo("identifier");
+        assertThat(property(changeset, "author")).isEqualTo("fbiville");
+        assertThat(property(changeset, "checksum")).isEqualTo(checksum(singletonList("CREATE (n: SomeNode {text:'yeah'})")));
     }
 
     private static void assertThatQueryIsExecuted(ResultSet resultSet) throws SQLException {
-        Node node = (Node) resultSet.getObject("node");
-        assertThat(node.getLabels()).containsExactly(DynamicLabel.label("SomeNode"));
-        assertThat(node.getProperty("text")).isEqualTo("yeah");
+        Object node = resultSet.getObject("node");
+        assertThat(property(node, "text")).isEqualTo("yeah");
     }
 
     private static void assertThatQueryIsNotExecuted(ResultSet resultSet) throws SQLException {
         assertThat(resultSet.getObject("node")).isNull();
     }
+
+    private static Object property(Object changeset, String name) {
+        if (changeset instanceof Map) {
+            return ((Map<String,String>) changeset).get(name);
```

Comment:
I think it's actually a `Map<String, Object>`, or `Map<String, ?>`. You could even say `Map<?, ?>`.